lib/Conversion/HWToBTOR2/HWToBTOR2.cpp

```diff
@@ -778,8 +800,27 @@ struct ConvertHWToBTOR2Pass
     StringRef regName = reg.getName().value();
     int64_t w = requireSort(reg.getType());
 
-    // Generate state instruction (represents the register declaration)
-    genState(reg, w, regName);
+    // Check for initial values which must be emitted before the state in btor2
+    Value pov = reg.getPowerOnValue();
+    if (pov) {
+      // Start by visiting the powerOnValue, we make the assumption that it will
+      // always be a constant and thus no operands need to be added to the
+      // worklist
+      dispatchTypeOpVisitor(pov.getDefiningOp());
```

Comment:
Why assume when it can easily be checked? `llvm::isa_and_nonull<hw::ConstantOp>` should do the trick and ideally produce a diagnostic message if it's unsupported.